sources/system/file-system/file-system.dylan

```diff
@@ -516,3 +516,25 @@ define sideways method list-locator
      locator);
   locators
 end method list-locator;
+
+///
+
+define generic create-symbolic-link
+  (target :: <pathname>, link :: <pathname>) => ();
+
+define method create-symbolic-link
+    (target :: <file-system-locator>, link :: <file-system-locator>) => ()
+  %create-symbolic-link(target, link)
+end method create-symbolic-link;
+
+define method create-symbolic-link
+    (target :: <string>, link :: <string>) => ()
+  create-symbolic-link(as(<file-system-locator>, target),
+                       as(<file-system-locator>, link))
+end method create-symbolic-link;
```

Comment:
Delete the above two methods, which are redundant with the method below.

Optional: since this is not an open generic, and the "dispatch" to OS-specific code is done via LID files, you could just define it with `define function`.

Reply:
You only want this function?
```
define function create-symbolic-link
    (target :: <pathname>, link :: <pathname>) => ()
  %create-symbolic-link(as(<file-system-locator>, target),
                        as(<file-system-locator>, link))
end function create-symbolic-link;
```
If the answer is yes, perhaps there is more code in this file that can be reviewed the same way.

Reply:
Right. Unless I'm very confused the other methods are totally redundant. And yes, I see other generic functions that could be turned into regular functions, but in a different PR.

One could argue that using specific methods is more correct because there's no need to call `as(<file-system-locator>, ...)` on arguments that are already locators, but I don't see how it's worth it.  Maybe @housel has a different opinion?

Reply:
Then I think the code is ready to merge.

Reply:
This is going to break the build on Win32. Even if we don't add support for actually creating symbolic links on Windows (which uses the `CreateSymbolicLink` API, which is not included in the link libraries we're using to build Open Dylan), we should at least leave a placeholder for `%create-symbolic-link` so that the compile will not fail.

Reply:
@housel Thank you. I added the placeholder in my last commit. Can you review it?